internal/controller/monovertexrollout/monovertexrollout_progressive.go

```diff
@@ -0,0 +1,35 @@
+package monovertexrollout
+
+import (
+	"context"
+
+	"github.com/numaproj/numaplane/internal/controller/progressive"
+	"github.com/numaproj/numaplane/internal/util/kubernetes"
+	"github.com/numaproj/numaplane/internal/util/logger"
+	apiv1 "github.com/numaproj/numaplane/pkg/apis/numaplane/v1alpha1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// TODO: can move other progressive interface functions here as well? (same with PipelineRollout, ISBServiceRollout)
+
+// make an assessment of the upgrading child to determine if it was successful, failed, or still not known
+// TODO: fix this assessment not to return an immediate result as soon as things are healthy or unhealthy
+func (r *MonoVertexRolloutReconciler) AssessUpgradingChild(ctx context.Context, existingUpgradingChildDef *unstructured.Unstructured) (apiv1.AssessmentResult, error) {
+	numaLogger := logger.FromContext(ctx)
+	upgradingObjectStatus, err := kubernetes.ParseStatus(existingUpgradingChildDef)
+	if err != nil {
+		return apiv1.AssessmentResultUnknown, err
+	}
+
+	numaLogger.
+		WithValues("namespace", existingUpgradingChildDef.GetNamespace(), "name", existingUpgradingChildDef.GetName()).
+		Debugf("Upgrading child is in phase %s", upgradingObjectStatus.Phase)
+
+	if upgradingObjectStatus.Phase == "Running" && progressive.IsNumaflowChildReady(&upgradingObjectStatus) {
+		return apiv1.AssessmentResultSuccess, nil
+	}
+	if upgradingObjectStatus.Phase == "Failed" {
+		return apiv1.AssessmentResultFailure, nil
+	}
+	return apiv1.AssessmentResultUnknown, nil
+}
```

Comment:
This used to be a single function in progressive.go, but it's now part of the progressiveController interface, and all 3 can have different implementations (although, if it makes sense they can use some shared function for common functionality).